go/vt/vttablet/tabletserver/throttle/throttler_test.go

```diff
@@ -75,3 +116,93 @@ func TestIsAppExempted(t *testing.T) {
 	throttler.UnthrottleApp("schema-tracker") // meaningless. App is statically exempted
 	assert.True(t, throttler.IsAppExempted("schema-tracker"))
 }
+
+func TestRefreshMySQLInventory(t *testing.T) {
+	metricsQuery := "select 1"
+	config.Settings().Stores.MySQL.Clusters = map[string]*config.MySQLClusterConfigurationSettings{
+		selfStoreName: {},
+		"ks1":         {},
+		"ks2":         {},
+	}
+	clusters := config.Settings().Stores.MySQL.Clusters
+	for _, s := range clusters {
+		s.MetricQuery = metricsQuery
+		s.ThrottleThreshold = &atomic.Uint64{}
+		s.ThrottleThreshold.Store(1)
+	}
+
+	throttler := &Throttler{
+		mysqlClusterProbesChan: make(chan *mysql.ClusterProbes),
+		mysqlClusterThresholds: cache.New(cache.NoExpiration, 0),
+		ts:                     &FakeTopoServer{},
+		mysqlInventory:         mysql.NewInventory(),
+	}
+	throttler.metricsQuery.Store(metricsQuery)
+	throttler.initThrottleTabletTypes()
+
+	validateClusterProbes := func(t *testing.T, ctx context.Context) {
+		testName := fmt.Sprintf("leader=%t", throttler.isLeader.Load())
+		t.Run(testName, func(t *testing.T) {
+			t.Run("waiting for probes", func(t *testing.T) {
+				ctx, cancel := context.WithTimeout(ctx, 10*time.Second)
+				defer cancel()
+				numClusterProbesResults := 0
+				for {
+					select {
+					case probes := <-throttler.mysqlClusterProbesChan:
```

Comment:
IMO it's worth a comment that we're NOT competing with the main `Operate()` goroutine on the channel read because that goroutine only reads from the channel if the tablet's throttler `IsOpen()` (which it's not in the test).

Reply:
Added comment.